components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.api.synchronizer/src/main/java/org/wso2/carbon/apimgt/micro/gateway/api/synchronizer/util/APIMappingUtil.java

```diff
@@ -0,0 +1,356 @@
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.FaultGatewaysException;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.api.model.APIStatus;
+import org.wso2.carbon.apimgt.api.model.CORSConfiguration;
+import org.wso2.carbon.apimgt.api.model.Scope;
+import org.wso2.carbon.apimgt.api.model.Tier;
+import org.wso2.carbon.apimgt.api.model.URITemplate;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import org.wso2.carbon.apimgt.impl.APIManagerFactory;
+import org.wso2.carbon.apimgt.impl.definitions.APIDefinitionFromOpenAPISpec;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.dto.APICorsConfigurationDTO;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.dto.APIDTO;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.dto.APIEndpointSecurityDTO;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.dto.APIMaxTpsDTO;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.exceptions.APISynchronizationException;
+import org.wso2.carbon.apimgt.micro.gateway.api.synchronizer.internal.ServiceDataHolder;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Util class for API creation
+ */
+public class APIMappingUtil {
+
+    private static final Log log = LogFactory.getLog(APIMappingUtil.class);
+
+    /**
+     * Method to update an existing API
+     *
+     * @param body DTO model of new API to be created
+     */
+    public static void apisUpdate(APIDTO body) throws APISynchronizationException {
+        try {
+            APIManagerConfiguration config = ServiceDataHolder.getInstance().
+                    getAPIManagerConfigurationService().getAPIManagerConfiguration();
+            String adminUsername = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_USERNAME);
+            String tenantDomain = MultitenantUtils.getTenantDomain(adminUsername);
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+
+            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(adminUsername);
+            boolean isWSApi = APIDTO.TypeEnum.WS == body.getType();
+
+            //Set admin as the API provider since admin is the only user in this tenant space
+
+            API apiToAdd = APIMappingUtil.fromDTOtoAPI(body, adminUsername);
+            APIIdentifier apiId = apiToAdd.getId();
+
+            // Delete the API if it already exists
+            if (apiProvider.checkIfAPIExists(apiId)) {
+                apiProvider.deleteAPI(apiId);
```

Comment:
What if API already has subscriptions?

So this still works as the users does not subscribe directly to the local API?

Reply:
The sole purpose of creating APIs this way is generating the synapse config xml of the API. If we have identical xml files in both environments, an identical response will be generated. So we do not perform any validations on parameters that do not go in the API's synapse configs.